scripts/run_auto-sklearn_for_metadata_generation.py

```diff
@@ -144,6 +145,7 @@
                                         include=include,
                                         metric=automl_arguments['metric'],
                                         cost_for_crash=get_cost_of_crash(automl_arguments['metric']),
+                                        port=logging.handlers.DEFAULT_TCP_LOGGING_PORT,
```

Comment:
I'm just seeing this and there won't be any server running there so actually log this. Do you think we can only tear down the logging server in the `__del__` instead of also tearing it down at the end of `fit()`?

Reply:
I think <__del__> is only called when you called del or when all references are removed right?

If for some reason any logger is still logging to the server, then references to the automl object will still be alive, so del would not be called. This can easily be triggered if you make the tae/exec in  smbo crash at start. Then the logger will never return if this logger clean up (with time out) doesn't trigger.

Do you mean this for the metalearning testing? Maybe it makes sense to make it so that if port is none we default to logging.getLogger with a warning message?

Reply:
> I think <del> is only called when you called del or when all references are removed right?

Yes, I think so, too.

> If for some reason any logger is still logging to the server, then references to the automl object will still be alive, so del would not be called. This can easily be triggered if you make the tae/exec in smbo crash at start. Then the logger will never return if this logger clean up (with time out) doesn't trigger.

I see.

> Do you mean this for the metalearning testing? Maybe it makes sense to make it so that if port is none we default to logging.getLogger with a warning message?

Both for testing and practical use. I think the idea of falling back to logging.getLogger is a great idea.